pkgs/applications/networking/browsers/chromium/update.py

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i python -p python3 -p nix
+
+import csv
+import json
+import re
+import subprocess
+from multiprocessing import Pool
+from codecs import iterdecode
+from os.path import dirname
+from urllib.request import urlopen
+
+DEB_URL = "https://dl.google.com/linux/chrome/deb/pool/main/g"
+BUCKET_URL = "https://commondatastorage.googleapis.com/chromium-browser-official";
+
+def nix_prefetch_url(url, hash="sha256"):
+    out = subprocess.check_output(["nix-prefetch-url", "--type", hash, url])
+    return out.decode('utf-8').rstrip()
+
+def prefetch_source(name, channel):
+    url = f'{BUCKET_URL}/chromium-{channel["version"]}.tar.xz'
+    return (name, {**channel, 'sha256': nix_prefetch_url(url)})
+
+def prefetch_bin(name, channel):
+    suffix = 'unstable' if name == 'dev' else name
+    longname = f'google-chrome-{suffix}'
+    url = f'{DEB_URL}/{longname}/{longname}_{channel["version"]}-1_amd64.deb'
+    return (name, {**channel, 'sha256bin64': nix_prefetch_url(url)})
+
+channels = {}
+
+with urlopen('https://omahaproxy.appspot.com/all?os=linux') as resp:
+    versions = csv.DictReader(iterdecode(resp, 'utf-8'))
+    for version in versions:
+        channels[version['channel']] = {'version': version['current_version']}
+
+with Pool(5) as pool:
+    # A good future enhancement would be to fetch both sources and
+    # binaries at the same.  The tasks aren't dependent on each other
+    # -- it's just more annoying to assemble the final channels
+    # dictionary that way.
+    channels = dict(pool.starmap(prefetch_source, channels.items()))
```

Comment:
Given that `upstream-info.json` is not checked before, this will *always* download all the tarballs even though no update is needed, so I'd suggest reading `upstream-info.json` and only download tarballs if there is a new version available.